wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Reaction/Handler.js

```diff
@@ -0,0 +1,375 @@
+		UiDialog,    UiUserList,        User,                 CountButtons, 
+		UiAlignment, UiCloseOverlay,    UiScreen
+	)
+	{
+		"use strict";
+		
+		/**
+		 * @constructor
+		 */
+		function UiReactionHandler(objectType, options) { this.init(objectType, options); }
+		UiReactionHandler.prototype = {
+			/**
+			 * Initializes the reaction handler.
+			 *
+			 * @param	{string}	objectType	object type
+			 * @param	{object}	options		initialization options
+			 */
+			init: function(objectType, options) {
+				if (options.containerSelector === '') {
+					throw new Error("[WoltLabSuite/Core/Ui/Reaction/Handler] Expected a non-empty string for option 'containerSelector'.");
+				}
+				
+				this._containers = new Dictionary();
+				this._details = new ObjectMap();
+				this._objectType = objectType;
+				this._cache = new Dictionary();
+				
+				this._popoverCurrentObjectId = 0;
+				
+				this._popover = null; 
+				
+				this._options = Core.extend({
+					// permissions
+					canReact: false,
+					canReactToOwnContent: false,
+					
+					// selectors
+					buttonSelector: '.reactButton', 
+					containerSelector: '',
+					isButtonGroupNavigation: false,
+					
+					// other stuff
+					parameters: {
+						data: {}
+					}
+				}, options);
+				
+				this.initReactButtons(options, objectType);
+				
+				this.countButtons = new CountButtons(this._objectType, this._options); 
+				
+				DomChangeListener.add('WoltLabSuite/Core/Ui/Reaction/Handler-' + objectType, this.initReactButtons.bind(this));
+				UiCloseOverlay.add('WoltLabSuite/Core/Ui/Reaction/Handler', this._closePopover.bind(this));
+			},
+			
+			/**
+			 * Initializes all applicable react buttons with the given selector.
+			 */
+			initReactButtons: function() {
+				var element, elements = elBySelAll(this._options.containerSelector), elementData, triggerChange = false, objectId;
+				for (var i = 0, length = elements.length; i < length; i++) {
+					element = elements[i];
+					objectId = ~~elData(element, 'object-id');
+					if (this._containers.has(objectId)) {
+						continue;
+					}
+					
+					elementData = {
+						reactButton: null,
+						objectId: ~~elData(element, 'object-id'), 
+						element: element
+					};
+					
+					this._containers.set(objectId, elementData);
+					this._initReactButton(element, elementData);
+					
+					triggerChange = true;
+				}
+				
+				if (triggerChange) {
+					DomChangeListener.trigger();
+				}
+			},
+			
+			
+			/**
+			 * Initializes a specific react button.
+			 */
+			_initReactButton: function(element, elementData) {
+				elementData.reactButton = elBySel(this._options.buttonSelector, element);
+				
+				if (elementData.reactButton === null || elementData.reactButton.length === 0) {
+					// the element may have no react button 
+					return; 
+				}
+				
+				if (elementData.reactButton.parentElement.parentElement.classList.contains('jsMobileNavigation')) {
```

Comment:
`.parentElement.parentElement` is oddly specific. Maybe `.closest` might be more readable.